annotation-server/src/gene-phenotypes/gene-phenotypes.service.ts

```diff
@@ -23,62 +25,89 @@ export class GenePhenotypesService {
         @InjectRepository(GenePhenotype)
         private genePhenotypeRepository: Repository<GenePhenotype>,
     ) {
-        this.hashedPhenotypes = new Map<string, Phenotype>();
+        this.hashedGeneSymbols = new Map();
+        this.hashedPhenotypes = new Map();
     }
 
-    private hashedPhenotypes: Map<string, Phenotype>;
-
     async fetchGenePhenotypes(): Promise<void> {
         await this.clearAllData();
 
         this.logger.log('Fetching gene-phenotype combinations from CPIC.');
         const response = this.httpService.get(
             'https://api.cpicpgx.org/v1/diplotype',
-            { params: { select: 'genesymbol, generesult' } },
+            {
+                params: {
+                    select: [
+                        'genesymbol',
+                        'generesult',
+                        'consultationtext',
+                    ].join(','),
+                },
+            },
         );
 
         const diplotypeDtos: DiplotypeDto[] = (await lastValueFrom(response))
             .data;
 
-        await this.geneSymbolRepository.save(
-            await this.getGeneSymbolDtos(diplotypeDtos),
+        await this.genePhenotypeRepository.save(
+            await this.getGenePhenotypes(diplotypeDtos),
         );
 
+        this.hashedGeneSymbols.clear();
         this.hashedPhenotypes.clear();
         this.logger.log(
             'Successfully saved gene-phenotype combinations to database.',
         );
     }
 
-    private async getGeneSymbolDtos(diplotypeDtos: DiplotypeDto[]) {
-        const genePhenotypes = new Map<string, Set<Phenotype>>();
+    private async getGenePhenotypes(
+        diplotypeDtos: DiplotypeDto[],
+    ): Promise<GenePhenotype[]> {
+        const genePhenotypes = new Map<string, GenePhenotype>();
```

Comment:
Sorry if this is too nitpicky but with how this map is used it would make more sense & be more efficient to substitute it with a set of key-strings and an array of `GenePhenotype`s.